python/cugraph/cugraph/dask/community/leiden.py

```diff
@@ -0,0 +1,160 @@
+import operator as op
+
+from pylibcugraph import ResourceHandle
+from pylibcugraph import leiden as pylibcugraph_leiden
+from typing import Tuple
+
+
+def convert_to_cudf(cupy_vertex, cupy_partition):
+    """
+    Creates a cudf DataFrame from cupy arrays from pylibcugraph wrapper
+    """
+    df = cudf.DataFrame()
+    df["vertex"] = cupy_vertex
+    df["partition"] = cupy_partition
+
+    return df
+
+
+def _call_plc_leiden(sID, mg_graph_x, max_iter, resolution, do_expensive_check):
+    return pylibcugraph_leiden(
+        resource_handle=ResourceHandle(Comms.get_handle(sID).getHandle()),
+        graph=mg_graph_x,
+        max_level=max_iter,
+        resolution=resolution,
+        do_expensive_check=do_expensive_check,
+    )
+
+
+def leiden(
+    input_graph, max_iter: int = 100, resolution: int = 1.0
+) -> Tuple[dask_cudf.DataFrame, float]:
+    """
+    Compute the modularity optimizing partition of the input graph using the
+    Leiden method
+
+    # FIXME: check if it is still the current implementation
+    It uses the Leiden method described in:
+
+    Traag, V. A., Waltman, L., & van Eck, N. J. (2019). From Louvain to Leiden:
+    guaranteeing well-connected communities. Scientific reports, 9(1), 5233.
+    doi: 10.1038/s41598-019-41695-z
+
+    Parameters
+    ----------
+    G : cugraph.Graph
+        The graph descriptor should contain the connectivity information
+        and weights. The adjacency list will be computed if not already
+        present.
+        The current implementation only supports undirected graphs.
+
+    max_iter : integer, optional (default=100)
+        This controls the maximum number of levels/iterations of the Leiden
+        algorithm. When specified the algorithm will terminate after no more
+        than the specified number of iterations. No error occurs when the
+        algorithm terminates early in this manner.
+
+    resolution: float, optional (default=1.0)
+        Called gamma in the modularity formula, this changes the size
+        of the communities.  Higher resolutions lead to more smaller
+        communities, lower resolutions lead to fewer larger communities.
+        Defaults to 1.
+
+    Returns
+    -------
+    parts : dask_cudf.DataFrame
+        GPU data frame of size V containing two columns the vertex id and the
+        partition id it is assigned to.
+
+        ddf['vertex'] : cudf.Series
+            Contains the vertex identifiers
+        ddf['partition'] : cudf.Series
+            Contains the partition assigned to the vertices
+
+    modularity_score : float
+        a floating point number containing the global modularity score of the
+        partitioning.
+
+    Examples
+    --------
+    >>> from cugraph.experimental.datasets import karate
+    >>> G = karate.get_graph(fetch=True)
+    >>> parts = cugraph.leiden(G)
+
+    """
+
+    if input_graph.is_directed():
+        raise ValueError("input graph must be undirected")
+
+    # Initialize dask client
```

Comment:
I think this comment could be worded better, since I didn't think it initialized the client.
```suggestion
    # Return a client if one has started
```